### PR TITLE
Fix cloud connection drops with >8 devices per broker

### PR DESCRIPTION
**Pull request by tomquist**
## Summary
Fixes an issue where accounts with more than 8 devices on a single broker would experience immediate connection drops, preventing any data from being exchanged. The remote brokers enforce two subscription limits that were not being respected:
1. A maximum of 8 topic filters per SUBSCRIBE packet
2. A maximum of 50 subscriptions per connection

## Changes
- **Added subscription limit constants and validation**: Exported `MAX_TOPICS_PER_SUBSCRIBE` (8) and `MAX_SUBSCRIPTIONS_PER_CONNECTION` (50) with detailed documentation explaining the broker behavior
- **Implemented `limitToSubscribable()` function**: Splits devices into those that fit within the subscription quota and those that exceed it
- **Configured MQTT client batching**: Set `subscribeBatchSize` to 8 to split SUBSCRIBE packets into sizes the broker accepts
- **Added device filtering in main startup**: Devices exceeding the per-broker quota are now filtered out upfront with a warning log listing the ignored devices, rather than failing silently
- **Added comprehensive test coverage**: Tests verify the quota enforcement, edge cases, and that full brokers can be subscribed to in valid packet batches

## Behavior
- Accounts with ≤50 devices per broker: No change, all devices forwarded
- Accounts with >50 devices per broker: First 50 devices are forwarded, remaining devices are logged and excluded
- All SUBSCRIBE packets are now batched to respect the 8 topic filter limit per packet

https://claude.ai/code/session_01FXfbsisq2knUbUQDXVX2Ls

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved cloud connectivity for accounts with more than 8 devices.
  * Accounts with more than 50 devices now forward the first 50 devices instead of failing completely.
  * Devices excluded due to connection limits are identified in warning logs.
  * Subscription requests are split into smaller batches to improve broker compatibility.

* **Documentation**
  * Added a changelog entry describing the device-forwarding improvements.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 ## [Next]
+- Fixed the cloud connection being dropped immediately, so no data was exchanged at all, when the account has more than 8 devices. Accounts with more than 50 devices now forward the first 50 and log which devices are left out, instead of failing entirely (#232)
```
[the remark this answers is not in the document]
Comment (tomquist) [reply CHANGELOG.md]: Correct — the cap is applied per broker, so an account whose devices land on more than one of them forwards more than 50. Reworded in d9df136, though not with the per-broker detail: which broker a device uses is derived from its model and firmware and is invisible to users, and this changelog is end-user facing (no internal names), so a "50 per broker" figure would not be something a reader could act on. It now says a large number of devices forwards as many as the cloud accepts and logs the ones left out — the log names them, which is the actionable part.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/mqtt_forwarder.test.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    // MAX_TOPICS_PER_SUBSCRIBE must keep every packet of a full broker legal.
+    const packetsBeforeLast = Math.floor(
+      MAX_SUBSCRIPTIONS_PER_CONNECTION / MAX_TOPICS_PER_SUBSCRIBE,
+    );
+    assert.ok(
+      packetsBeforeLast * MAX_TOPICS_PER_SUBSCRIBE <
+        MAX_SUBSCRIPTIONS_PER_CONNECTION,
+      "a full broker must not need a packet that starts at the quota",
+    );
+  });
```
[the remark this answers is not in the document]
Comment (tomquist) [reply src/mqtt_forwarder.test.ts:91]: Both halves were right, addressed in two commits.

The arithmetic bug: `Math.floor` gives the wrong starting count whenever the batch size divides the quota. It would have failed for a batch size of 10 (`5 * 10 = 50`) even though five packets of ten are legal. d9df136 derives the packet count first and then the subscriptions held before the last packet, which holds for every batch size from 1 to 50.

The missing regression guard: 1a8e5c4 added a test that fails if `subscribeBatchSize` is dropped, verified by deleting the line and watching it fail. It asserts on the options the remote client connects with rather than on captured packets — building a client to observe real packets means real PEM material in the test, and this repo gitignores `*.key`/`*.crt`, so committing a throwaway key to assert a constant seemed the worse trade. The end-to-end behaviour (batched SUBSCRIBEs on connect and on reconnect) was verified against the live brokers while developing the fix.

---
_Generated by [Claude Code](https://claude.ai/code)_
